subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyFactory.java

```diff
@@ -58,6 +58,16 @@ public interface DependencyFactory {
      */
     ExternalModuleDependency create(CharSequence dependencyNotation);
 
+    /**
+     * Create an {@link ExternalModuleDependency} from an id and a version string.
+     *
+     * @param id the id
+     * @param version the version (optional)
+     * @return the new dependency
+     * @since 8.1
+     */
+    ExternalModuleDependency create(String id, @Nullable String version);
+
```

Comment:
:x: It is not clear what `id` is. I would clarify this both in the method signature and its javadoc.
```suggestion
    /**
     * Create an {@link ExternalModuleDependency} from a plugin identifier and a version string.
     *
     * @param pluginId the plugin identifier
     * @param version the plugin version (optional)
     * @return the new dependency
     * @since 8.1
     */
    ExternalModuleDependency create(String pluginId, @Nullable String version);

```